microcosm_daemon/state_machine.py

```diff
@@ -50,5 +52,7 @@ def run(self):
             with self.graph.signal_handler:
                 while self.should_run():
                     self.current_state = self.step()
+                    if self.reloader:
```

Comment:
The event loop checks the reloader on every loop iteration (but only if `debug` is enabled).